ctools/defragment_sharded_collection.py

```diff
@@ -696,116 +687,127 @@ async def get_remain_chunk_imbalance(center, target_chunk):
                     continue
                 else:
                     break
+                
+            if total_shard_size[shard] * args.shard_imbalance_frac < max_shard_size and not center_size_kb == 0:
+                continue
 
             # chunks should be on other shards, but if this script was executed multiple times or 
             # due to parallelism the chunks might now be on the same shard            
 
             left_chunk = chunks_max_index.get(pickle.dumps(c['min']))
             right_chunk = chunks_min_index.get(pickle.dumps(c['max']))
             
-            # Exclude overweight target shards
-            if (left_chunk is not None and right_chunk is not None) and (left_chunk['shard'] != right_chunk['shard']):
-                if total_shard_size[left_chunk['shard']] > total_shard_size[right_chunk['shard']] * args.shard_imbalance_frac:
+            if left_chunk and right_chunk:
+                for dest_chunk in [left_chunk, right_chunk]:
```

Comment:
I would add some comment along the voting logic to explain the rationale behind each step

Reply:
Yes I'll do